src/services/ioSignService.ts

```diff
@@ -0,0 +1,121 @@
+  ResponseErrorNotFound,
+  ResponseErrorValidation,
+  ResponseSuccessJson
+} from "@pagopa/ts-commons/lib/responses";
+
+import { IoSignAPIClient } from "src/clients/io-sign";
+import { SignerDetailView } from "generated/io-sign-api/SignerDetailView";
+import { FilledDocumentDetailView } from "generated/io-sign-api/FilledDocumentDetailView";
+import {
+  EmailString,
+  FiscalCode,
+  NonEmptyString
+} from "@pagopa/ts-commons/lib/strings";
+import { Id } from "generated/io-sign-api/Id";
+
+import {
+  ResponseErrorStatusNotDefinedInSpec,
+  withCatchAsInternalError,
+  withValidatedOrInternalError
+} from "../utils/responses";
+import { readableProblem } from "../../src/utils/errorsFormatter";
+import { ResponseErrorNotFound403 } from "./eucovidcertService";
+
+export default class IoSignService {
+  constructor(private readonly ioSignApiClient: ReturnType<IoSignAPIClient>) {}
+
+  /**
+   * Get the Signer id related to the user.
+   */
+  public readonly getSignerByFiscalCode = (
+    fiscalCode: FiscalCode
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<SignerDetailView>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.getSignerByFiscalCode({
+        body: { fiscal_code: fiscalCode }
+      });
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 200:
+            return ResponseSuccessJson(response.value);
+          case 400:
+            return ResponseErrorValidation(
+              "Invalid request",
+              `An error occurred while validating the request body | ${response.value}`
+            );
+          case 403:
+            return ResponseErrorNotFound403(
+              "The user associated with this profile could not be found."
+            );
+          case 500:
+            return ResponseErrorInternal(
+              `Internal server error | ${response.value}`
+            );
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
+        }
+      });
+    });
+
+  public readonly createFilledDocument = (
+    document_url: NonEmptyString,
+    email: EmailString,
+    family_name: NonEmptyString,
+    name: NonEmptyString,
+    signerId: Id
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<FilledDocumentDetailView>
+  > =>
+    withCatchAsInternalError(async () => {
+      const validated = await this.ioSignApiClient.createFilledDocument({
+        body: {
+          document_url,
+          email,
+          family_name,
+          name
+        },
+        "x-iosign-signer-id": signerId
+      });
+      return withValidatedOrInternalError(validated, response => {
+        switch (response.status) {
+          case 201:
+            return ResponseSuccessJson(response.value);
```

Comment:
We have `IResponseSuccessRedirectToResource` if needed here

Reply:
There is no need because a redirect should not be done. The endpoint will be saved and later called by the app.

Reply:
Ok, maybe in this case you've to change io-sign api specs don't you? :)

Reply:
Maybe you are getting confused with 301 which is redirect? ☺️

This API returns a `201` (created) but I haven't found an `IResponse` with this status code. So it's not a problem to convert it to `200` backend side and leave `201` function side (it's legit).
Or do you have any suggestions?

Reply:
Nope. Accordingly to https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201 a `201` Response should contain a location where the client can poll for resource information if needed. Maybe redirect is a "too much" term applied in this case :), so if we don't need the location info, why using 201 instead of 200 or 202 ? It's not a problem on backend's side that works as a proxy, it's a general consideration 😄

Reply:
See this for example https://github.com/pagopa/io-backend/blob/ad8264d062f25f8ecf6aa1cc9755eb3cf413593c/src/controllers/cgnController.ts#L96

Reply:
Furthermore, `200` is not consistent with the status code exposed in the openapi (`201`)

Reply:
You are definitely right! I proceeded to change the `io-sign` code to also return the `Location` field in the header and aligned the `io-backend` code. 
Thanks ☺️